include/avr/pgmspace.h

```diff
@@ -95,8 +95,18 @@
 #endif
 
 #ifndef __ATTR_PROGMEM__
+#if defined __has_attribute
+#if __has_attribute(__progmem__)
```

Comment:
That won't work either when `!__has_attribute`, because then `PROGMEM` will not be defined.  Why not just leave the GCC stuff untouched and do you clang stuff? Like:
```
#ifndef __ATTR_PROGMEM__
#ifdef __clang_version__
/* do clang stuff */
#else
#define __ATTR_PROGMEM__ __attribute__((__progmem__))
#endif
#endif
```